### PR TITLE
Improvements to documentation

### PR DESCRIPTION
**Pull request by tiberiuichim**
Looking for feedback.

An overview of the biggest changes:

- I've removed the "Customization" section, 
- I've renamed "Recipes" to "Development recipes", it also receives all the files in "Customization"
- I've added a new section, "Addons". It has the old content for addons from the Customization
- A new page in Addons, "Best practices" for addons
- I've made "Getting started" a section. The old page becomes "Bootstrap Volto"
- I've added a new page "Developer roadmap" under "Getting started"
- I've also added "Learning resources" under "Getting started", with links to training and Youtube videos

### CODE REVIEW

**Comment by tiberiuichim** [comment]
@plone/volto-team It would be nice to get this reviewed and merged, there are quite a few improvements to the docs, there's a lot of interest in the community around Volto, it's important to have nicer docs, if possible.

File: docs/source/addons/index.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+An addon can be almost anything that a Volto project can be. They can:
+
+- provide additional views and blocks
+- override or extend Volto's builtin views, blocks, settings
+- shadow (customize) Volto's (or another addon's) modules
+- register custom routes
+- provide custom Redux actions and reducers
+- register custom Express middleware for Volto's server process
+- tweak Volto's webpak configuration, load custom Razzle and Webpack plugins
```
Comment (sneridagh) [inline docs/source/addons/index.md:24]: typo

File: docs/source/addons/index.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+You can install a Volto addon just like any other JS package:
+
+```
+yarn add name-of-addon
+```
+
+If the addon is not published on NPM, you can retrieve it directly from Github:
+
+```
+yarn add collective/volto-dropdownmenu
```
Comment (sneridagh) [inline docs/source/addons/index.md:38]: should we reference the new org?   🙂

File: docs/source/addons/index.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    "collective-another-volto-addon"
+  ],
+  ...
+}
+```
+
+!!! warning
+    Adding the addon package to the `addons` key is obligatory! It allows Volto
+    to treat that package properly and provide it with BabelJS language
+    features. It is the equivalent of including a Python egg to the `zcml`
```
Comment (sneridagh) [inline docs/source/addons/index.md:72]: maybe "in Plone terminology" or in "Plone world"

File: docs/source/addons/index.md (hunk 2 above)
Comment (tiberiuichim) [reply docs/source/addons/index.md:38]: plone-collective?

**Comment by tisto** [comment]
I square read the PR and this is indeed a great improvement. Just a general recommendation for our technical docs. We should try to write as short as possible and avoid rationalizations and long explanations. We do not need to rationalize the decisions that we made and we can just state things as they are. Most people scan dev documentation instead of reading it from the start to the end. This should be reflected in the structure and writing style of our dev docs. Maybe @svx has to add something here as well. :)

**Review by tisto — APPROVED** [review]
+1 of merging this right away. We can always refine later. 

At some point, we should try to invite @svx to a nice place somewhere in the world to give all of us training on how to write good technical docs.

**Comment by svx** [comment]
> +1 of merging this right away. We can always refine later.
> 
> At some point, we should try to invite @svx to a nice place somewhere in the world to give all of us training on how to write good technical docs.

Would be glad to !

**Pull request merged**